haystack/components/converters/openapi_functions.py

```diff
@@ -83,13 +86,10 @@ def run(self, sources: List[Union[str, Path, ByteStream]], system_messages: List
                 try:
                     service_openapi_spec = self._parse_openapi_spec(openapi_spec_content)
                     functions: List[Dict[str, Any]] = self._openapi_to_functions(service_openapi_spec)
-                    docs = [
-                        Document(
-                            content=json.dumps(function),
-                            meta={"spec": service_openapi_spec, "system_message": system_message},
-                        )
-                        for function in functions
-                    ]
+                    meta_dict: Dict[str, Any] = {"spec": service_openapi_spec}
+                    if system_message:
+                        meta_dict["system_message"] = system_message
+                    docs = [Document(content=json.dumps(function), meta=meta_dict) for function in functions]
```

Comment:
Just to be sure, you're sharing the same `meta_dict` instance across all the documents, this might have unexpected behaviour downstream if some of those documents have their metadata changed.

Reply:
No, that's a mistake, sharp 👀

Reply:
ok should be corrected now @masci see [2ce1049](https://github.com/deepset-ai/haystack/pull/6825/commits/2ce104964912c4187fe5739dfe0862ce8a0ed941) for more details